graviti/operation/frame.py

```diff
@@ -123,8 +138,22 @@ def do(  # pylint: disable=invalid-name
             sheet: The sheet name.
 
         """
-        data = self.data.to_pylist()
-        for batch in chunked(data, _MAX_BATCH_SIZE):
+        dataframe = self.data
+        arrays = _get_file_arrays(dataframe.schema, dataframe)  # type: ignore[arg-type]
+        data = dataframe.to_pylist()
+
+        for batch, *file_arrays in chunked(zip(data, *arrays), _MAX_BATCH_SIZE):
```

Comment:
will it be clear to separate binary file upload and add data into two for loop？

Reply:
Do you mean uploading sheet data after done uploading all the binary files? 
I think that would make the upload operation seem slow.

Reply:
OK we can refine it afterwards